### PR TITLE
feat: search common component | tags

### LINKED ISSUE #711: [FEATURE] Tags & Search Component | selection of tags
### Detailed description

Tags & Search Component - 
- remove the link with tags, 
-  make tags active when selected

### Context

more intuitive way.

### Possible implementation

- [ ] remove the link with tags - don't show the name of the tag in search box
- [ ] make tags active when selected - when user select a tag, make it orange (selected one), if multiple tags, make multiple in orange

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Adamss10**
## Fixes Issue
Closes #711 and #745 

## Changes proposed

- Tags are no longer displayed in search box when selected
- Added colors to selected tags 
- Wrapped tags in blogs and feed in a Set (to remove duplicates) 

## Screenshots

image

## Note to reviewers

Although i tried my best to test my changes it may still need some thorogh testing , as the components i changed are used in many places . I tested the feeds , blogs and explore page 

## Code of Conduct

- [x] By submitting this pull request, I confirm I've read and complied with the [CoC](https://github.com/thecyberworld/Support/blob/main/CODE_OF_CONDUCT.md) 🖖

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

---

You can also join our [Discord](https://discord.gg/QHBPq6xP5p) community.
Feel free to check out other cool repositories of the [Thecyberworld](https://github.com/thecyberworld).
Join the Thecyberworld GitHub Organisation by raising an [issue](https://github.com/thecyberworld/Support/issues/new?assignees=&labels=invite+me+to+the+organisation&template=invitation.yml&title=Please+invite+me+to+the+GitHub+Community+Organization) (you will be sent an invitation).

### CODE REVIEW

**Comment by Adamss10** [comment]
@ArkadiK94 Is there something wrong ?

**Comment by ArkadiK94** [comment]
- [ ]  you don't check for capitalisation in the set, I created feed with "feed" tag and "Feed" tag and both appering
image

- [ ] when I pick the tag "feeds" and after a tag which includes in the tag "feeds" like "feed" it doesn't work properly, and doesn't show the right feed.
- [ ] when the tag includes more then one word it doesn't work properly

https://github.com/thecyberworld/TheCyberHUB/assets/76536506/4002c9ee-81d8-4c68-86b1-d1295d7febf3

- [ ] when I pick tag "feed" and then wants to search for "feed1" it removed this text automatically when I write "feed" , hmm, it should wait for me to finish typing, and toast error, or not remove it if it comes from the search box input


https://github.com/thecyberworld/TheCyberHUB/assets/76536506/984638cc-5637-49ea-81b2-9d8fcc88cb47

File: src/components/Common/SearchInputBox/SearchInputBox.jsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React, { useRef, useState } from "react";
 import { SearchBox, SearchIcon, SearchInput, CancelIcon, CancelIconWrapper } from "./SearchInputBoxElements";
 import { FcSearch } from "react-icons/fc";
 
-const SearchInputBox = ({ placeholder, value, onChange, setValue }) => {
+const SearchInputBox = ({ placeholder, value, onChange, clearValue }) => {
```
Comment (ArkadiK94) [inline src/components/Common/SearchInputBox/SearchInputBox.jsx:5]: Every component that placed in Common folder, if you make a change you responsible to check where this component was used and change there accordingly.

https://github.com/thecyberworld/TheCyberHUB/assets/76536506/0636e80f-782d-486d-9353-61dc81ca2a7f

In future we plan to add storybook or similar for those components but still for now you should be aware that the components in this folder can be used in many places at the app.

File: src/components/Feeds/FeedTags/SocialTags.jsx [hunk 2]
```diff
@@ -1,14 +1,31 @@
...
+    };
     return (
         <TagsContainer>
             <AllTags>
                 {tags.slice(0, 10).map(
                     (tag, key) =>
                         tag.length !== 0 && (
-                            <Tag key={key} onClick={() => handleClick(tag)}>
+                            <Tag
+                                style={{
```
Comment (ArkadiK94) [inline src/components/Feeds/FeedTags/SocialTags.jsx:20]: Why didn't you add those styles inside the 'Tag' styled-component?

**Review by ArkadiK94 — CHANGES_REQUESTED** [review]
@Adamss10 Hi, thanks for your contribution, I have some comments, please fix them :)

File: src/components/Feeds/FeedTags/SocialTags.jsx (hunk 2 above)
Comment (Adamss10) [reply src/components/Feeds/FeedTags/SocialTags.jsx:20]: i thought maybe since an override i put it inline . I should place in the styled component ?

Comment (ArkadiK94) [reply src/components/Feeds/FeedTags/SocialTags.jsx:20]: https://styled-components.com/docs/basics#passed-props

**Comment by Adamss10** [comment]
hey , 
i have refactored the way filtering by tags works because it was a bit problematic and led me to go for workarounds that are buggy  : 

- initially the tag filters and the search text were managed by a single state "searchTerm" 
- Now i created 2 seperate states for tags and searchterm .
-  Along the way there's some code that is no longer needed (it deals with invisible characters due to appending tags alongside the searchterm which i think is unecessary ) .
- I tried to not break stuff elsewhere but please let me know if something's wrong .

**Review by ArkadiK94 — APPROVED** [review]
@Adamss10 Looks Great, thanks for your implementations and refactoring 💯

**Pull request merged**